src/Integration/LocalServices/ErrorListInfoBarController.cs

```diff
@@ -255,38 +256,58 @@ private void UpdateRequired(string customInfoBarMessage = null)
 
         private void CurrentErrorWindowInfoBar_Closed(object sender, EventArgs e)
         {
-            this.ClearCurrentInfoBar();
+            // Called on the UI thread -> must handle exceptions
```

Comment:
There are two other event handlers for which uncaught exceptions (will cause VS to crash - clicking on the button in the infobar, and closing the infobar.

I haven't changed the existing code, just wrapped it in a try-catch block.

I couldn't sensibly add a test for this click handler. The handler only calls one injectable object, but that object is only created after a series of stateful transitions (i.e. checking that an update is required, then creating and wiring up an infobar).

The test would either be extremely complex, or I'd have to do quite a bit of refactoring of the product code. Writing the test didn't seem like a good use of time, and changing the product code is too risky.

I've manually tested both code paths by debugging and changing the code to throw an exception at an appropriate point.